browser/ui/whats_new/whats_new_util.cc

```diff
@@ -0,0 +1,142 @@
+/* Copyright (c) 2023 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/browser/ui/whats_new/whats_new_util.h"
+
+#include <string>
+
+#include "base/metrics/field_trial_params.h"
+#include "base/ranges/algorithm.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/strings/stringprintf.h"
+#include "base/version.h"
+#include "brave/browser/ui/whats_new/pref_names.h"
+#include "brave/components/l10n/common/locale_util.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_tabstrip.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/version_info/version_info.h"
+#include "url/gurl.h"
+
+namespace {
+
+double g_testing_major_version = 0;
+constexpr char kWhatsNewTrial[] = "WhatsNewStudy";
+constexpr char kWhatsNewTargetMajorVersion[] = "target_major_version";
+
+absl::optional<double> GetTargetMajorVersion() {
+  const std::string target_major_version_string = base::GetFieldTrialParamValue(
+      kWhatsNewTrial, kWhatsNewTargetMajorVersion);
+  // Field trial doesn't have this value.
+  if (target_major_version_string.empty()) {
+    return absl::nullopt;
+  }
+
+  double target_major_version;
+  if (!base::StringToDouble(target_major_version_string,
+                            &target_major_version)) {
+    return absl::nullopt;
+  }
+
+  return target_major_version;
+}
+
+// Returns 1.xx or 2.xx as double.
+absl::optional<double> GetCurrentBrowserVersion() {
```

Comment:
why do we need all of this when the chromium code already does this on major version updates?